src/navigation/failure_identification/failure_identification.py

```diff
@@ -122,7 +129,9 @@ def update(self, nav_status: SmachContainerStatus, drive_status: MotorsStatus, o
         # create a new row for the data frame
         self.actively_collecting = True
         cur_row = {}
-        cur_row["time"] = rospy.Time.now()
+        cur_row["row"] = self.row_counter
```

Comment:
I don't think we need "row" I think we can just use iloc like df[i:j] to get the dataframe from rows i to j

Reply:
When we tried to index into the dataframe before without adding the specific row column, it didn't work because the indices for each row are all 0. We tried working with pandas.DataFrame.reset_index but found that creating our own row column and indexing into the dataframe using this was the easiest.

Reply:
ok yeah I think we've had this conversation before, sorry I totally forgot. I just played around with this a bit and I see what you're saying, when you turn the new row into a data frame the row index gets initialized to zero so you just end up with a bunch of zero index rows. but I think I fixed this problem by doing `pd.concat([df,DataFrame([cur_row])],ignore_index=True)`, which seems to reset the indices correctly. have you guys tried this? seems like reset index does the same thing except it adds a new column for your old indices which is not what you want.

This might not be the most important thing to work on right now before URC, but the reason I'm pushing for it is because it seems pretty silly to make our own index column when there's already one built in, especially with all the extra work we have to do to slice it. I haven't thoroughly tested my solution so I could be wrong, but I feel like it should be possible to use the builtin row index

Reply:
also, I tested this little more and it seems like iloc indexing doesn't actually care about the role indices, and it behaves correctly even if they're all zero, which is why iloc works in our current code